src/Engine/Events/EventMap.h

```diff
@@ -2,16 +2,23 @@
 
 #include <unordered_map>
 #include <vector>
+#include <string>
 
 #include "Engine/Events/EventIR.h"
 
 class EventMap {
  public:
-    void add(int id, EventIR ir);
+    void add(int eventId, EventIR ir);
     void clear() { _eventsById.clear(); }
 
+    void execute(int eventId, int startStep, bool canShowMessages) const;
+    std::string getHintString(int eventId) const;
+
     void dumpAll() const;
-    void dump(int id) const;
+    void dump(int eventId) const;
  private:
     std::unordered_map<int, std::vector<EventIR>> _eventsById;
 };
+
+void eventProcessor(int eventId, int targetObj, int canShowMessages, int startStep);
+std::string getEventHintString(unsigned int eventId);
```

Comment:
These functions don't really belong here?

They know about event map locality, and this knowledge is in Engine.h. EventMap knows nothing about event locality, so these two functions aren't part of the extended EventMap API.